src/pint/models/pulsar_binary.py

```diff
@@ -185,7 +187,33 @@ def apply_units(self):
             bparObj.value = bparObj.value * u.Unit(bparObj.units)
 
     def update_binary_object(self, toas=None, acc_delay=None):
-        """Update binary object instance for this set of parameters/toas."""
+        """Update stand alone binary's parameters and toas from PINT-facing object.
+
+        This function passes the PINT-facing object's parameter values and TOAs
+        to the stand-alone binary object. If the TOAs are not provided, it only
+        updates the parameters not the TOAs.
+
+        Parameters
+        ----------
+        toas: `~pint.toa.TOAs` object, optional
+            The TOAs that need to pass to the stand alone model.Default value is
+            None. If toas is None, this function only updates the parameter value.
+            If 'acc_delay' is not provided, the stand alone binary receives the
+            standard barycentered TOAs.
+
+        acc_delay: `~numpy.ndarray`
+            If provided, TOAs will be corrected by provided acc_delay instead of
+            the standard barycentering. The stand alone binary receives the
+            input TOAs - acc_delay.
+
+        Warns
+        -----
+        If the 'toas' are not provided, the stand alone binary model will use
+        the TOAs were passed to it from last interation (i.e. last barycnetered
+        TOAs) or no TOAs for stand alone binary model at all. This behavior will
+        cause incorrect answers. Allowing optional 'toas' input is for some
+        lower level tests. We do not recommend PINT user to use it.
```

Comment:
Just to be clear: the recommendation is that PINT users always pass TOAs, right? Perhaps removing the default value of None would help ensure this - users who know they need to can still explicitly pass None.